### ISSUE TITLE
RCS1143 probably false alarm for functions without arguments

### ISSUE THREAD

**Issue by devowl** [JosefPihrt/Roslynator#773]
Hello, i have RCS1143 warning on. But looks like it's false alarm in case of code design below (when you need to have several retries in row until first success)

```
public static void Main()
{
    var result = Test() ?? Test();
}

private static string Test()
{
    try
    {
        using (WebClient client = new WebClient())
        {
            return client.DownloadString("http://someurl.url");
        }
    }
    catch
    {
        return null;
    }
}
```

My fix was code below:

```
var result = Test();
result ??= Test();
result ??= Test();
```

I don't know is it bug or not.
Thanks

**Comment by JosefPihrt** [comment]
I would say that the diagnostic should be reported in this case.

The question is what to do if second attempt fails. So maybe there should be  a loop:
```cs
public static void Main()
{
    string result = null;

    while ((result = Test()) == null)
    {
    }
}
```

Or you can suppress the diagnostic.
```cs
#pragma warning disable RCS1143
var result = Test() ?? Test();
#pragma warning restore RCS1143
```

**Comment by devowl** [comment]
Okay, not a problem then, thanks for you answer
